_posts/en/newsletters/2023-03-08-newsletter.md

```diff
@@ -0,0 +1,205 @@
+        opcode a commitment to those conditions in the form of a hash
+        digest.
+
+    - *Alice initiates an unvaulting* by spending the output received to
+      the above taproot tree (using it as an input) and choosing the
+      trigger tapscript.  At this time, she provides two additional
+      parameters to the `OP_TRIGGER_FORWARD` opcode, the index of the
+      output which will receive this input's funds and a hash-based
+      commitment to how she wants to be able to spend the funds later.
+      The opcode verifies that the indicated output of this transaction
+      pays a taproot tree similar to the one being spent except that the
+      trigger tapscript is replaced with a script using an
+      `OP_CHECKSEQUENCEVERIFY` (CSV) relative delay equal to the delay
+      specified previously (e.g., 1000 blocks) and an
+      `OP_FORWARD_OUTPUTS` opcode which includes Alice's commitment
+      hash.  The method of reconstructing the taproot tree is similar to
+      an earlier [covenant][topic covenants] proposal,
+      `OP_TAPLEAF_UPDATE_VERIFY` (see [Newsletter #166][news166 tluv]).
+
+    - *Alice completes the unvaulting* by waiting until the relative
+      timelock has expired and then spending the unvaulting output,
+      choosing the tapleaf with the `OP_FORWARD_OUTPUTS` opcode.  The
+      opcode verifies that the spending transaction's output amounts and
+      scripts hash to the same commitment Alice made in the previous
+      transaction.  In this case, Alice has successfully deposited funds
+      to a vault, begun an unvaulting, been forced to wait at least
+      1,000 blocks to allow her monitoring programs to verify she really
+      did want to spend the funds to the specified outputs, and
+      completed the spend.
+
+    - If something goes wrong, *Alice freezes the funds*.  She can do
+      this at any time up until an unvaulting is completed by simply
+      choosing to spend using the freeze tapscript from the output of
+      either the vaulting or initial unvaulting transactions.  Recall
+      that Alice explicitly placed the freeze tapscript in the vaulting
+      transaction, and note that it was implicitly carried over by the
+      transaction with initiated the unvaulting.
+
+  One of the advantages to users of this approach over the original
+  `OP_VAULT` design is that the freeze tapscript can contain any
+  authorization conditions Alice wants to specify.  In the `OP_VAULT`
+  proposal, anyone knowing the parameters chosen by Alice could spend
+  her funds to the freeze script.  That wasn't a security problem but it
+  could be annoying.  In Sanders's design, Alice could (for example)
+  require a signature from a very lightly protected wallet in order to
+  initiate a freeze---this would perhaps be enough of a burden to
+  prevent most griefing attacks but not enough of a barrier to prevent
+  Alice from quickly freezing her funds in an emergency.
```

Comment:
```suggestion
  One of the advantages to users of this approach over the original
  `OP_VAULT` design is that the freeze leafscript can contain any
  authorization conditions Alice wants to specify.  In the `OP_VAULT`
  proposal, anyone knowing the parameters chosen by Alice could spend
  her funds to the freeze script.  That wasn't a security problem but it
  could be annoying.  In Sanders's design, Alice could (for example)
  require a signature from a very lightly protected wallet in order to
  initiate a freeze---this would perhaps be enough of a burden to
  prevent most griefing attacks but not enough of a barrier to prevent
  Alice from quickly freezing her funds in an emergency.
```